bin/lib/library_builder.py

```diff
@@ -767,18 +768,24 @@ def get_build_annotations(self, buildfolder):
             return json.loads(buffer)
 
     def get_commit_hash(self) -> str:
```

Comment:
Might be more trouble than it's worth but tou could consider a `@lrucache` which does maybe what you need here? (that is, caches and if nothing else changes it would return the same value. ( https://docs.python.org/3/library/functools.html )

Reply:
caches are a bad idea here, it matters that before this the nightly install happens and libs may or may not have been refreshed

Reply:
The cache is in-memory. It's similar to what you've implemented manually here; but honestly whatever works :)